openlibrary/templates/account/reading_log.html

```diff
@@ -71,7 +71,21 @@
       $:macros.Pager(int(current_page), doc_count, results_per_page=results_per_page)
   $else:
     <ul class="list-books">
-      $if checkin_year:
+      $if q:
+        <form method="GET" class="olform pagesearchbox">
+          <input type="text" minlength="3" placeholder="$_('Search your reading log')" name="q" value="$(query_param('q', ''))"/>
+          <input type="submit"/>
+        </form>
+        $ query = query_param('q')
+        <center>
+          <hr>
+          <div class="red">$_("No results found in %s" %key) </div>
```

Comment:
```suggestion
          <div class="red">$_("No results found in this shelf") </div>
```
The value of `key` is not suitable for the UI, and no UI strings for shelf names are readily available in this template.  Let's change the message to something that works more generally.